torchvision/prototype/models/quantization/shufflenetv2.py

```diff
@@ -24,18 +24,20 @@
 ]
 
 
+@handle_num_categories_mismatch()
 def _shufflenetv2(
     stages_repeats: List[int],
     stages_out_channels: List[int],
+    *,
     weights: Optional[Weights],
     progress: bool,
     quantize: bool,
     **kwargs: Any,
 ) -> QuantizableShuffleNetV2:
-    if weights is not None:
-        _ovewrite_named_param(kwargs, "num_classes", len(weights.meta["categories"]))
-        if "backend" in weights.meta:
-            _ovewrite_named_param(kwargs, "backend", weights.meta["backend"])
+    # FIXME
+    # if weights is not None:
+    # if "backend" in weights.meta:
+    #     _ovewrite_named_param(kwargs, "backend", weights.meta["backend"])
```

Comment:
Is this a one-time-thing or should we have another decorator for that?

Reply:
This is again part of the building mechanism logic. I wouldn't move this to a decorator. Also it's not a one-time thing, it's quite common for all quantization models.